resourcemanager/commonids/composite_resource_id.go

```diff
@@ -0,0 +1,68 @@
+package commonids
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+)
+
+type CompositeResourceID[T1 resourceids.ResourceId, T2 resourceids.ResourceId] struct {
+	First  T1
+	Second T2
+}
+
+func (id CompositeResourceID[T1, T2]) ID() string {
+	fmtString := "%s|%s"
+	return fmt.Sprintf(fmtString, id.First.ID(), id.Second.ID())
+}
+
+func (id CompositeResourceID[T1, T2]) String() string {
+	fmtString := "%s\n%s"
+	return fmt.Sprintf(fmtString, id.First.String(), id.Second.String())
+}
+
+func ParseCompositeResourceID[T1 resourceids.ResourceId, T2 resourceids.ResourceId](input string, first T1, second T2) (*CompositeResourceID[T1, T2], error) {
+	return Parse(input, first, second, false)
+}
+
+func ParseCompositeResourceIDInsensitively[T1 resourceids.ResourceId, T2 resourceids.ResourceId](input string, first T1, second T2) (*CompositeResourceID[T1, T2], error) {
```

Comment:
could we document this struct/the functions to explain how this is intended to be used?